homeassistant/components/onvif/camera.py

```diff
@@ -111,6 +114,24 @@
         DOMAIN, SERVICE_PTZ, async_handle_ptz, schema=SERVICE_PTZ_SCHEMA
     )
 
+    async def async_handle_reboot(service):
+        """Handle PTZ service call."""
+        all_cameras = hass.data[ONVIF_DATA][ENTITIES]
+        entity_ids = await async_extract_entity_ids(hass, service)
+        target_cameras = []
+        if not entity_ids:
+            target_cameras = all_cameras
+        else:
+            target_cameras = [
+                camera for camera in all_cameras if camera.entity_id in entity_ids
+            ]
+        for camera in target_cameras:
+            await camera.async_perform_reboot()
```

Comment:
Can we speed this up by awaiting all camera calls concurrently, eg by using `asyncio.gather`?

Reply:
possible i guess, but fairly unlikely if you ask me. To be honest: I'd improve it but don't know how `asyncio,gather` works.

Reply:
Awaiting `asyncio.gather(*coros)` will schedule all coroutines on the event loop and return when all are done. Exceptions are either propagated immediately or returned in the result list.

https://docs.python.org/3/library/asyncio-task.html#asyncio.gather

Awaiting tasks in loops should be avoided if the task order doesn't matter and there's not a problem with executing the tasks "at the same time". Some devices or APIs can't handle this.